### ISSUE TITLE
Impossible to add new rule (says folder not found)

### ISSUE THREAD

**Issue by Sbaz18** [opto/nostalgy-xpi#76]
If I try to modify or add a new rule, the folder to move to is found, and I can select it, but.... when I click OK, I get the error message "[Javascript application] Please choose an existing folder".

This makes the app useless, as I have moved all my folder structure.

**Comment by opto** [comment]
thanks for the info. Which TB version, which OS?

Klaus

**Comment by opto** [comment]
if you remove the >> text before the actual folder name, then it works here

**Comment by Sbaz18** [comment]
> thanks for the info. Which TB version, which OS?
> 
> Klaus

On Mac (latest version : Big Sur.
TB latest too (78.6.1 (64-bit))

**Comment by Sbaz18** [comment]
> if you remove the >> text before the actual folder name, then it works here

thans, but it makes no change when I just input the folder name. Still the same message.

**Comment by jimbeauxh** [comment]
I'm new to nostalgy++, but as I was upgrading to TB78 I got the same error when I tried to edit an existing rule. The problem was that the "under folder" was empty. I'm guessing that somehow under TB68 I had entered a rule that did not have the "under folder" populated or it got cleared accidentally.

**Comment by Sbaz18** [comment]
Thank you @jimbeauxh for your workaround. The under folder used to be optional, which was better. I have several email addresses (implying several inboxes) that similar people use, and this means I must now replicate each rule 5 times.
I don't see any added value in making this field compulsory.

**Comment by DavidR2PA** [comment]
Thunderbird 78.7.0 (64-bit)
Windows 10
Nostalgy 2.2.4

I also am having trouble with making rules.
I can't make a rule without an Under Folder" location, but I don't want to put one in and none of my older rules have one.
I want the rule to work no matter where the message is, but usually it's in one of my 10 inboxes.
Nostalgy just complains and say "Please choose an existing folder"

So, I figured I'd Export / Import after adding a rule
Nope - says it does but nothing changes.

So, I figured I screwed it up so I exported / changed one rule / Imported
Nope - says it does but nothing changes.

Apparently I can't make a rule without a folder and the Import is broken.

**Comment by Sbaz18** [comment]
Yep, the more I try it , the more I think the take over of Nostalgy has just broken it and we should move to an alternative. 

Iza

**Comment by DavidR2PA** [comment]
Nope - Nostalgy is too good to lose. If it goes away I'll go to outlook. :(
-------- Original message --------From: Sbaz18 <notifications@github.com> Date: 2/1/21  6:15 PM  (GMT-05:00) To: opto/nostalgy-xpi <nostalgy-xpi@noreply.github.com> Cc: DavidR2PA <DavidR@Fast.net>, Comment <comment@noreply.github.com> Subject: Re: [opto/nostalgy-xpi] Impossible to add new rule (says folder not found) (#76) 

Yep, the more I try it , the more I think the take over of Nostalgy has just broken it and we should move to an alternative. 



Iza

**Comment by Olifair** [comment]
@Sbaz18 Well, Nostalgy was broken due to the changes in TB, not due a "takeover" and for me it works seamlessly, so I don't know who "we" is, but I'm clearly very happy to be able to use my preferred keyboard shortcuts again and move easily my emails from my inbox to my hundreds of folders ;-)

**Comment by opto** [comment]
personally, I would find operation without "under" folder convenient as well.
I was never aware that did work in older versions - the code looks as if it is required.

However, whatever is happening now is done by TB. 

Just as the fact that pressing esc in the input box doesn't input an esc character (ASCII 27) into the box, but it enters the characters 'esc' into the box. (aka workaround needed for new TB). It is 'fun' to figure out whether the folder string contains the letters esc (intentionally), or whether this was caused by pressing esc key.

So it seems we need another workaround for 'under'. I will mark this as bug - with respect to old TB - and as enhancement - with respect to what current TB allows out of the box. TB 78 removed textbox and introduced html:input instead, and that causes what you observe.

Klaus

**Comment by DavidR2PA** [comment]
Would it be easier to fix the Import funtionality?

When I export them, most of my rules look like this, and they work just fine:
{sender:true,recipients:true,subject:false,contains:"yarde",under:"",folder:"Local\u0020Folders/\u0020DISCUS/Vendors/Yarde"}

Notice the  ,under:"", section.

If I were able to Export, add a few lines and Import it would be great!
Export works, adding or changing lines is easy but Import does not work...

**Comment by opto** [comment]
that is good news, it means that the moving code still accepts the empty under, and that I 'only' need to override the input box so that it accepts 'no entry'.

If easy, I will do that around this weekend.

Klaus

**Comment by DavidR2PA** [comment]
If that's not "easy" perhaps
        making Import work?
        I wonder if Import does not work because of the empty under
        field...
      
    On 2/2/2021 11:27 AM, opto wrote:
    
    
      
      that is good news, it means that the moving code still accepts
        the empty under, and that I 'only' need to override the input
        box so that it accepts 'no entry'.
      If easy, I will do that around this weekend.
      Klaus
      —
        You are receiving this because you commented.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/opto/nostalgy-xpi/issues/76#issuecomment-771761821",
"url": "https://github.com/opto/nostalgy-xpi/issues/76#issuecomment-771761821",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by DavidR2PA** [comment]
Is it also possible to remove
        or option off the H++ Help button on the status bar to reduce
        the space Nostalgy takes up, perhaps so it won't make the
        program window so much wider?  Help is available elsewhere...
    
    On 2/2/2021 11:27 AM, opto wrote:
    
    
      
      that is good news, it means that the moving code still accepts
        the empty under, and that I 'only' need to override the input
        box so that it accepts 'no entry'.
      If easy, I will do that around this weekend.
      Klaus
      —
        You are receiving this because you commented.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/opto/nostalgy-xpi/issues/76#issuecomment-771761821",
"url": "https://github.com/opto/nostalgy-xpi/issues/76#issuecomment-771761821",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by opto** [comment]
this is working now. Under must be empty, then the rule is applied everywhere.
The old code used something that isn't documented any longer for javascript, so I took it out. Apparently, even if undocumented, it is still working, as you report. I now replaced it with a valid expression.

Later today, I will put a release 2.2.5 under releases here in github.

Klaus

**Comment by Sbaz18** [comment]
Great news @opto ! thanks a lot for debugging this point and to all who have helped understanding this annoying behaviour !

Iza

**Comment by opto** [comment]
https://github.com/opto/nostalgy-xpi/releases/tag/v2.2.5

**Comment by DavidR2PA** [comment]
Thank you VERY much !!!
    
    On 2/5/2021 7:31 AM, opto wrote:
    
    
      
      this is working now. Under must be empty, then the rule is
        applied everywhere.
        The old code used something that isn't documented any longer for
        javascript, so I took it out. Apparently, even if undocumented,
        it is still working, as you report. I now replaced it with a
        valid expression.
      Later today, I will put a release 2.2.5 under releases here in
        github.
      Klaus
      —
        You are receiving this because you commented.
        Reply to this email directly, view it on GitHub, or unsubscribe.
      [
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/opto/nostalgy-xpi/issues/76#issuecomment-774005101",
"url": "https://github.com/opto/nostalgy-xpi/issues/76#issuecomment-774005101",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by DavidR2PA** [comment]
WORKS !!!

I noticed this:
New functions: set width of folderbox in prefs

Pleae advise where this is and how I can get to it?

**Comment by DavidR2PA** [comment]
OK, I found the folderbox width in prefs and it worked until I used Nostalgy at which time the window expanded to the right a whole lot and did not come back.  Is this a known bug?
